File: app/src/main/java/com/hfut/schedule/logic/utils/ClipBoard.java
package com.hfut.schedule.logic.utils;


import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;

import com.hfut.schedule.App.MyApplication;

public class ClipBoard {
    public static void copy(String s) {
        // 获取系统剪贴板
        ClipboardManager clipboard = (ClipboardManager) MyApplication.context.getSystemService(Context.CLIPBOARD_SERVICE);
        // 创建一个剪贴数据集，包含一个普通文本数据条目（需要复制的数据）
        ClipData clipData = ClipData.newPlainText(null, s);
        // 把数据集设置（复制）到剪贴板
        clipboard.setPrimaryClip(clipData);
    }
}

File: app/src/main/java/com/hfut/schedule/logic/utils/ClipBoard.kt
package com.hfut.schedule.logic.utils

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import com.hfut.schedule.App.MyApplication
import com.hfut.schedule.ui.utils.components.showToast

object ClipBoard {
    @JvmStatic
    fun copy(str : String,tip : String = "已复制到剪切板") {
        try {
            val clipboard = MyApplication.context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val clipData = ClipData.newPlainText(null, str)
            clipboard.setPrimaryClip(clipData)
            showToast(tip)
        } catch (e:Exception) {
            showToast("复制到剪切板失败")
        }
    }
    @JvmStatic
    fun paste(): String {
        return try {
            val clipboard = MyApplication.context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            if (clipboard.hasPrimaryClip() && clipboard.primaryClip!!.itemCount > 0) {
                clipboard.primaryClip!!.getItemAt(0).text?.toString() ?: ""
            } else {
                ""
            }
        } catch (e: Exception) {
            showToast("获取剪切板内容失败")
            ""
        }
    }
}